flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/declarative/StateTrackingMockExecutionGraph.java

```diff
@@ -0,0 +1,459 @@
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void failGlobal(Throwable t) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public JobStatus waitUntilTerminal() throws InterruptedException {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean transitionState(JobStatus current, JobStatus newState) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void incrementRestarts() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void initFailureCause(Throwable t) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void vertexFinished() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void vertexUnFinished() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void notifyPartitionDataAvailable(ResultPartitionID partitionId) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Map<ExecutionAttemptID, Execution> getRegisteredExecutions() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void updateAccumulators(AccumulatorSnapshot accumulatorSnapshot) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void registerJobStatusListener(JobStatusListener listener) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void assertRunningInJobMasterMainThread() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void notifySchedulerNgAboutInternalTaskFailure(
+            ExecutionAttemptID attemptId,
+            Throwable t,
+            boolean cancelTask,
+            boolean releasePartitions) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public ShuffleMaster<?> getShuffleMaster() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public JobMasterPartitionTracker getPartitionTracker() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public ResultPartitionAvailabilityChecker getResultPartitionAvailabilityChecker() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public PartitionReleaseStrategy getPartitionReleaseStrategy() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public ExecutionDeploymentListener getExecutionDeploymentListener() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void deregisterExecution(Execution exec) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void notifyExecutionChange(Execution execution, ExecutionState newExecutionState) {
+        throw new UnsupportedOperationException();
```

Comment:
given that updateTaskExecutionState is a safe operation, I'd say this one should be too.